packs/ansible/actions/lib/ansible_base.py

```diff
@@ -0,0 +1,63 @@
+import os
+import sys
+import subprocess
+import shell
+
+__all__ = [
+    'AnsibleBaseRunner'
+]
+
+
+class AnsibleBaseRunner(object):
+    """
+    Base class for all Ansible Runners
+    """
+    BINARY_NAME = None
+    REPLACEMENT_RULES = None
+
+    def __init__(self, args):
+        """
+        :param args: Input command line arguments
+        :type args: ``list``
+        """
+        self.args = args[1:]
+
+    def execute(self):
+        """
+        Execute the command and stream stdout and stderr output
+        from child process as it appears without delay.
+        Terminate with child's exit code.
+        """
+        exit_code = subprocess.call(self.cmd, env=os.environ.copy())
+        if exit_code is not 0:
+            sys.stderr.write('Executed command "%s"\n' % ' '.join(self.cmd))
+        sys.exit(exit_code)
+
+    @property
+    @shell.replace_args('REPLACEMENT_RULES')
+    def cmd(self):
+        """
+        Get full command line as list.
+
+        :return: Command line.
+        :rtype: ``list``
+        """
+        return [self.binary] + self.args
+
+    @property
+    def binary(self):
+        """
+        Get full path to executable binary located in pack virtualenv.
+
+        :return: Full path to executable binary.
+        :rtype: ``str``
+        """
+        if not self.BINARY_NAME:
+            sys.stderr.write('Ansible binary file name was not specified')
+            sys.exit(1)
+        path = os.path.join('/opt/stackstorm/virtualenvs/ansible/bin', self.BINARY_NAME)
```

Comment:
Think it'd be too terribly difficult to expose this as an ENV variable for shell runners? I could see this (and that pattern) coming in handy